### PR TITLE
Fix copyrights

### PR DESCRIPTION
**Pull request by tmshort**
Add copyright to files that were missing it.
Update license from OpenSSL to Apache as needed.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Review by mattcaswell — DISMISSED** [review]

**Comment by mattcaswell** [comment]
Oh....hmmm....CI failure looks relevant, so my approval was premature

**Comment by tmshort** [comment]
Damn you `make update`!

**Comment by tmshort** [comment]
@mattcaswell, now it's happy

File: include/openssl/obj_mac.h [hunk 1]
```diff
@@ -5447,6 +5447,14 @@
...
 
 #endif /* OPENSSL_OBJ_MAC_H */
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
```
Comment (mattcaswell) [inline include/openssl/obj_mac.h:5457]: This seems an odd place to be adding a copyright - and it looks to me like this file already has a copyright at the beginning???

Comment (tmshort) [reply include/openssl/obj_mac.h:5457]: Yes, this diff is in the fixup commit. It was done by `make update`, and it's omission caused the CI to fail. I don't know why it was inserted by `make update`. This PR was supposed to be copyrights-only via comments. I was trying to avoid any actual code changes...

Comment (t8m) [reply include/openssl/obj_mac.h:5457]: That's because the `make update` copies the obj_compat.h here verbatim including the copyright comment from the file.

Comment (tmshort) [reply include/openssl/obj_mac.h:5457]: I figured it was something like that. And obj_compat.h was updated in this PR to include a missing copyright.

File: crypto/objects/obj_compat.h [hunk 2]
```diff
@@ -1,3 +1,11 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
```
Comment (mattcaswell) [inline crypto/objects/obj_compat.h:8]: This is the cause of the "make update" weirdness. This file is not actually a standalone headerfile. obj_mac.h gets auto generated by objects.pl, and then this file gets concatenated onto the end of it.

So, I'm thinking that we should revert the change in this file?

Comment (tmshort) [reply crypto/objects/obj_compat.h:8]: Then there'd be no copyright in that file...

Comment (bernd-edlinger) [reply crypto/objects/obj_compat.h:8]: so what?

Comment (tmshort) [reply crypto/objects/obj_compat.h:8]: It means that file is not subject to the Apache License, with all subsequent ramifications, thereof.
The project wanted to be under the Apache License 2.0. One could argue that if there's no license on a file, that someone could subsequently put a different license on a file, including GPL, and all that entails.

Comment (bernd-edlinger) [reply crypto/objects/obj_compat.h:8]: but **this** is a super-ugly hack:
`Configurations/unix-Makefile.tmpl:	( cd $(SRCDIR);	cat crypto/objects/obj_compat.h >> include/openssl/obj_mac.h )`
why don't we just do this `echo #include "obj_compat.h" >> include/openssl/obj_mac.h )`, and move that file
to include/openssl/ ?

Comment (bernd-edlinger) [reply crypto/objects/obj_compat.h:8]: for a different PR of course.

Comment (tmshort) [reply crypto/objects/obj_compat.h:8]: I agree. But the extra comment doesn't hurt. However, adding/moving a new header file changes things a bit more. I know a few build systems that would choke on a new header file being added (not that they couldn't be fixed).

Comment (bernd-edlinger) [reply crypto/objects/obj_compat.h:8]: ... or just do this `	( cd $(SRCDIR); grep "^#" crypto/objects/obj_compat.h >> include/openssl/obj_mac.h )`

Comment (mattcaswell) [reply crypto/objects/obj_compat.h:8]: Or `( cd $(SRCDIR); tail +8 crypto/objects/obj_compat.h >> include/openssl/obj_mac.h )`

Comment (bernd-edlinger) [reply crypto/objects/obj_compat.h:8]: note that `tail +8` does not work on all versions:
`'`
$ tail +8                             
tail: cannot open '+8' for reading: No such file or directory
$ tail --version 
tail (GNU coreutils) 8.21
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Paul Rubin, David MacKenzie, Ian Lance Taylor,
and Jim Meyering.
```

Comment (mattcaswell) [reply crypto/objects/obj_compat.h:8]: I think you might need to instead do `tail -n +8` on some versions (works with my version too)

The same can be achieved using sed/awk etc, e.g. see https://unix.stackexchange.com/questions/37790/how-do-i-delete-the-first-n-lines-of-an-ascii-file-using-shell-commands

Comment (tmshort) [reply crypto/objects/obj_compat.h:8]: It does not appear that `awk` is used outside of `util/perl/OpenSSL/config.pm` but `sed` is used inside `Configurations/unix-Makefile.tmpl`, so that seems a bit safer. `tail` is also not used in the build system (AFAICT), and I don't want to introduce any new system dependencies.

Comment (tmshort) [reply crypto/objects/obj_compat.h:8]: Fixed with a `sed -e '1,8d'`, which removes the first 8 lines of the file, which is *just* the copyright. So it should not impact further modification of the file.

File: include/openssl/obj_mac.h (hunk 1 above)
Comment (tmshort) [reply include/openssl/obj_mac.h]: Dealt with in the `unix-Makefile.tmpl` file

**Review by mattcaswell — APPROVED** [review]
Approved assuming the CIs still pass.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master and 3.0 branches. Thank you for your contribution.

File: crypto/objects/obj_compat.h (hunk 2 above)
Comment (levitte) [reply crypto/objects/obj_compat.h:8]: It'll be interesting if we change this boilerplate some day, enough to add, say, one extra line :wink:
